FinancialTracker/AuthManager.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  AuthHandler.swift
+//  FinancialTracker
+//
+//  Created by Boyan Pavlov on 5.02.22.
+//
+
+import Foundation
+import Firebase
+import FirebaseAuth
+
+class AuthManager {
+    private let auth = Auth.auth()
+    private(set) var databaseManager: DatabaseManager = DatabaseManager()
+    
+    init() {
+        self.databaseManager.setAuthManager(authManager: self)
```

Comment:
If you want to go really with injection, your `init` function for the `AuthManager` should take in the properties that you have initialised inline. For example:
```swift
 init(auth:..., databaseManager: ... ) {
         self.auth = Auth.auth()
         self.databaseManager = DatabaseManager()
         // The `setAuthManager` of `DatabaseManager` may not be a function but be part of the `init` for the `DatabaseManager`
         self.databaseManager.setAuthManager(authManager: self)
}
```